### PR TITLE
Tganley/bugfix/payload len

### PR DESCRIPTION
**Pull request by tganley**

### CODE REVIEW

File: src/beaconDecoder.py [hunk 1]
```diff
@@ -18,8 +27,29 @@ def run(self):
         beacon_port = 1
         payload_offset = 16
         packet_num_offset = 4
-        decoded_data = base64.b64decode(self.rawdata)
-        decoded_data = bytearray(decoded_data)
+
+        # Check if socket is still open
+        try:
+            self.s.send(bytes("You good bro?", 'utf-8'))
```
Comment (rcunrau) [inline src/beaconDecoder.py:33]: Are you sure this is a good idea? I thought there were calls like stat(2) to non-invasively check socket status.

Comment (rcunrau) [reply src/beaconDecoder.py:33]: Besides which, self.parser should have warned you on receive if there was a problem

**Review by rcunrau — CHANGES_REQUESTED** [review]
I see the PR is titled fix payload length, but there's certainly lots of other interesting things going on :-) I guess the actual payload fix is at line 46?

File: src/beaconDecoder.py (hunk 1 above)
Comment (tganley) [reply src/beaconDecoder.py:33]: I couldn't figure out how to do it non-invasively

Comment (tganley) [reply src/beaconDecoder.py:33]: Figured it out now - but moved the changes to another branch

**Comment by tganley** [comment]
I am going to close this PR and replace it with #200 which is more properly named and has further changes

**Pull request closed without merging**